File: easy/1422_maximum_score_after_splitting_a_string/solution.cpp
#include <numeric>
#include <string>
#include <vector>

class Solution {
public:
  int maxScore(std::string s) {
    const int sLen = s.size();

    std::vector<int> sList;

    for (const char &ch : s)
      sList.push_back(ch - '0');

    int left = sList[0] == 0 ? 1 : 0;
    int right = reduce(sList.begin(), sList.end()) - sList[0];

    int maxScore = 0;

    for (int i = 0; i < sLen - 1; ++i) {
      maxScore = std::max(left + right, maxScore);

      left += sList[i + 1] == 0 ? 1 : 0;
      right -= sList[i + 1];
    }

    return maxScore;
  }
};

File: easy/1422_maximum_score_after_splitting_a_string/solution.py
class Solution:
    def maxScore(self, s: str) -> int:
        s = list(map(int, s))

        left = 1 if s[0] == 0 else 0
        right = sum(s) - s[0]

        max_score = 0

        for i in range(len(s) - 1):
            max_score = max(left + right, max_score)

            left += 1 if s[i + 1] == 0 else 0
            right -= s[i + 1]

        return max_score
